tracer/src/Datadog.Trace/Agent/Api.cs

```diff
@@ -140,15 +183,72 @@ public async Task<bool> SendTracesAsync(ArraySegment<byte> traces, int numberOfT
                     continue;
                 }
 
-                _log.Debug<int>("Successfully sent {Count} traces to the DD agent", numberOfTraces);
                 return true;
             }
         }
 
-        private async Task<bool> SendTracesAsync(ArraySegment<byte> traces, int numberOfTraces, IApiRequest request, bool finalTry)
+        private async Task<bool> SendStatsAsyncImpl(IApiRequest request, bool isFinalTry, SendStatsState state)
+        {
+            bool success = false;
+
+            // Set additional headers
+            if (_containerId != null)
+            {
+                request.AddHeader(AgentHttpHeaderNames.ContainerId, _containerId);
+            }
+
+            using var stream = new MemoryStream();
+            state.Stats.Serialize(stream, state.Duration);
+
+            var buffer = stream.GetBuffer();
+
+            using var response = await request.PostAsync(new ArraySegment<byte>(buffer, 0, (int)stream.Length), MimeTypes.MsgPack).ConfigureAwait(false);
+
+            if (response.StatusCode is >= 200 and < 300)
+            {
+                success = true;
+            }
+            else if (isFinalTry)
+            {
+                try
+                {
+                    var responseContent = await response.ReadAsStringAsync().ConfigureAwait(false);
+                    _log.Error<int, string>("Failed to submit stats with status code {StatusCode} and message: {ResponseContent}", response.StatusCode, responseContent);
```

Comment:
This sounds a little like you failed to submit only the stats that have a specific status code and message.
```suggestion
                    _log.Error<int, string>("Failed to submit stats. Status code {StatusCode}, message: {ResponseContent}", response.StatusCode, responseContent);
```